internal/pkg/aws/ecs/ecs.go

```diff
@@ -39,6 +46,9 @@ type ssmSessionStarter interface {
 type ECS struct {
 	client         api
 	newSessStarter func() ssmSessionStarter
+
+	// Overrriden by unit test
+	tryNum int
```

Comment:
I think we should do the opposite here and not store **mutable state** in the structs. So for example,
```go
type ECS struct {
...

	maxForceDeploymentTries int
	pollIntervalDuration *time.Duration
}
```
This way in the constructor you can set these values to the constants, and in the tests you can override these two values while not having to worry about resetting `tryNum` after invocation. It also allows the methods to be called in parallel.

Reply:
that's a good idea! good call!